src/net_processing.cpp

```diff
@@ -1398,6 +1398,14 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             LogPrint("net", "Unparseable reject message received\n");
         }
 
+        if (strMsg == NetMsgType::BLOCK) {
+            // The node requested a block from us and then rejected it, which indicates that it's most likely running
+            // on rules which are incompatible to ours. Better to ban him after some time as it might otherwise keep
+            // asking for the same block (if -addnode/-connect was used on the other side).
+            LOCK(cs_main);
+            Misbehaving(pfrom->id, 10);
```

Comment:
May be it is better to use 5 or even 1 as banscore here? Node has only 10 request attempts before ban now, it can spend all of them installing wrong version and re-installing other version after that,  for example.

Reply:
Yeah, that might be a problem, but even with a ban score of 1 it will happen pretty fast that the node gets banned from everywhere. I'm considering only doing a disconnect now.

Reply:
But at the same time, this problem only occurs for nodes that use `addnode` and they will only be banned from the specified nodes. But still, a ban score of 1 might be better here.